samples/web/content/testrtc/ui/line-chart.html

```diff
@@ -0,0 +1,119 @@
+<!--
+ *  Copyright (c) 2014 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+-->
+
+<!--
+`<line-chart>` is used to display a line chart updating in real-time to the user
+in an efficient way and without locking the js thread for too long.
+
+It is not a very advanced or a pretty line chart render, but it is very specialized
+for the testrtc use case where delaying the JS thread incurs delays on measuring
+the round-trip time.
+-->
+<link rel="import" href="../../polymer/polymer.html">
+<polymer-element name="line-chart">
+  <template>
+    <div style="width:300px; margin:auto;">
+      <canvas id="canvas" width="240" height="150"></canvas>
+    </div>
+  </template>
+  <script>
+    var interval = 0;
+    Polymer({
+      x: [],
+      y: [],
+      yScale: [0, 600],
+      yLabels: [0, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500],
+      xStart: null,
+      xOffset: null,
+      lastSlowRender: null,
+      addDatapoint: function (timestamp, value) {
+        var start = window.performance.now();
+        this.ctx = this.$.canvas.getContext('2d');
+
+        var index = this.x.length;
+        this.x.push(timestamp / 100.0 * 2);
+        this.y.push(this.calculateY(value));
+
+        if (this.xStart === null || this.x[index] - this.xStart > this.$.canvas.width) {
+          // First point added or a point added that clears the chart.
+          this.setupChart();
+          this.xStart = this.x[index] - this.xOffset;
+        } else {
+          this.ctx.strokeStyle = (Math.min(this.y[index], this.y[index-1]) < 50 ? 'red': 'green');
+          this.ctx.beginPath();
+          this.ctx.moveTo(this.x[index - 1] - this.xStart, this.y[index - 1]);
+          this.ctx.lineTo(this.x[index] - this.xStart, this.y[index]);
+          this.ctx.stroke();
+          this.ctx.closePath();
+        }
+
+        var now = window.performance.now();
+        var interval = now - start;
+        if (interval > 5 && (this.lastSlowRender === null || (now - this.lastSlowRender) > 5000)) {
+          this.lastSlowRender = now;
+          console.log('Updating the graph was slow and might affect results. Graph update took ' + interval + ' ms');
+        }
+      },
+
+      setupChart: function () {
+        this.ctx.clearRect (0 , 0, this.$.canvas.width, this.$.canvas.height);
+        this.ctx.lineWidth = window.devicePixelRatio;
+        this.ctx.save();
+
+        this.ctx.lineWidth = window.devicePixelRatio;
+        this.ctx.strokeStyle = 'lightgray';
```

Comment:
For styles it should be 3 hex color codes where possible but since this is an html file it's fine.